service/test/unit/resources/test_account_recovery_resource.py

```diff
@@ -46,10 +44,65 @@ def assert_200_when_user_logged_in(_):
     def test_post_returns_successfully(self):
         request = DummyRequest(['/account-recovery'])
         request.method = 'POST'
+        self.resource._handle_post = MagicMock(return_value=defer.succeed(None))
+
         d = self.web.get(request)
 
         def assert_successful_response(_):
             self.assertEqual(200, request.responseCode)
 
         d.addCallback(assert_successful_response)
         return d
+
+    def test_post_returns_failure(self):
+        request = DummyRequest(['/account-recovery'])
+        request.method = 'POST'
+        self.resource._handle_post = MagicMock(return_value=defer.fail(InvalidPasswordError))
```

Comment:
We are using a mock based approach around private methods to test.  It does cover all scenarios but shouldn't it be just sufficient to test the public method and not test each private method thats called internally?

And because of this approach, we end up having a need for an 'integration' test of sorts which tests the complete flow and not just individual methods.